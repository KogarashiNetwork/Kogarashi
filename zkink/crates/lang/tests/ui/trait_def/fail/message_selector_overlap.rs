use ink_lang as ink;

#[ink::trait_definition]
pub trait TraitDefinition {
    #[ink(message, selector = 1)]
    fn message_1(&self);

    #[ink(message, selector = 1)]
    fn message_2(&self);
}

fn main() {}
