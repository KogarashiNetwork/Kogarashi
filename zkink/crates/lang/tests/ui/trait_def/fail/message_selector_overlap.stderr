error: encountered duplicate selector ([0, 0, 0, 1]) in the same ink! trait definition
 --> tests/ui/trait_def/fail/message_selector_overlap.rs:9:8
  |
9 |     fn message_2(&self);
  |        ^^^^^^^^^

error: first ink! trait constructor or message with same selector found here
 --> tests/ui/trait_def/fail/message_selector_overlap.rs:6:8
  |
6 |     fn message_1(&self);
  |        ^^^^^^^^^
